app/client/src/RouteChangeListener.tsx

```diff
@@ -0,0 +1,16 @@
+import { routeChanged } from "actions/focusHistoryActions";
+import React from "react";
+import { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { useLocation } from "react-router-dom";
+import { AppsmithLocationState } from "utils/history";
+
+export default function RouteChangeListener() {
+  const location = useLocation<AppsmithLocationState>();
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(routeChanged(location));
+  }, [location.pathname, location.hash]);
+  // eslint-disable-next-line react/jsx-no-useless-fragment
+  return <></>;
```

Comment:
Return Null instead 
```suggestion
  return null;
```

Explanation
https://reactjs.org/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored